src/TRoomDB.cpp

```diff
@@ -51,11 +51,7 @@ TRoom* TRoomDB::getRoom(int id)
     if (id < 0) {
         return nullptr;
     }
-    const auto i = rooms.constFind(id);
-    if (i != rooms.constEnd() && i.key() == id) {
-        return i.value();
-    }
-    return nullptr;
+    return rooms.value(id, nullptr);
```

Comment:
The second argument is unneeded in this usage - https://doc.qt.io/qt-5/qhash.html#value :
```suggestion
    return rooms.value(id);
```

Reply:
I've seen that link and I am unsure why you think this. The default is not a null pointer but a default-constructed value.

Reply:
A default constructed value in this case ***is a `nullptr`***! Remember that the `QHash` is a `<int, TRoom*>`...